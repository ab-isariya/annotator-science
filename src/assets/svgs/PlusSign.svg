<svg width="16" height="16" fill="none" xmlns="http://www.w3.org/2000/svg">
  <path fill-rule="evenodd" clip-rule="evenodd" d="M15.2 8.8H8.8v6.4a.8.8 0 01-1.6 0V8.8H.8a.8.8 0 110-1.6h6.4V.8a.8.8 0 111.6 0v6.4h6.4a.8.8 0 010 1.6z" fill="currentColor"/>
</svg>
