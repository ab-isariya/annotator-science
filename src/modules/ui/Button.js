/**
 * Buttons
 * figma: https://www.figma.com/file/d2iT8aM5wR0AN6ZHukOu5C/TAT-Style-Sheet-and-Master-Components?node-id=1986%3A17
 */

import PropTypes from 'prop-types';

import classnames from 'classnames';

/**
 * Icon displaying the status of a given annotation. Note this component
 * is not positioned: the parent component must define where it goes
 */
const Button = ({
  children,
  onClick,
  red,
  dark,
  vertical,
  secondary,
  tertiary,
  blue,
  small,
  large,
  icon,
  disabled,
  className,
  ...otherProps
}) => {
  //NOTE(Rejon): We use classnames to intelligently choose which name to use for typing.
  //             this is so we can improve DX around button usage without creating multiple buttons
  //             or requiring prop="string" for type, alt, and size.
  const type =
    classnames({
      secondary,
      tertiary
    }) || 'primary';

  const alt =
    classnames({
      red,
      dark,
      vertical,
      blue
    }) || null;

  const size =
    classnames({
      small,
      large
    }) || 'medium';

  const primaryButtonClass = classnames({
    //Primary
    'bg-blue-500': !alt,
    'text-white': !disabled,
    'hover:bg-blue-600 focus:bg-blue-600 focus:drop-shadow-focus-button':
      !alt && !disabled,
    'disabled:bg-grey-100 disabled:text-grey-300': alt !== 'dark' && disabled,
    //Primary - Red
    'bg-red-600': alt === 'red',
    'hover:bg-red-700 focus:bg-red-700 focus:drop-shadow-focus-red':
      alt === 'red' && !disabled,
    //Primary - Dark
    'bg-grey-600': alt === 'dark',
    'hover:bg-grey-500 focus:bg-grey-500 focus:drop-shadow-focus-dark':
      alt === 'dark' && !disabled,
    'disabled:bg-grey-700 disabled:text-grey-400': alt === 'dark' && disabled
  });

  const secondaryButtonClass = classnames('bg-white border', {
    'border-grey-500 hover:bg-grey-50 focus:bg-grey-50 focus:drop-shadow-focus-button':
      !disabled,
    'disabled:text-grey-200 disabled:border-grey-200': disabled,
    'rounded-md px-1.5 py-1.5 flex flex-col items-center justify-center text-xs gap-y-1':
      alt === 'vertical'
  });

  const tertiaryButtonClass = classnames({
    //Tertiary
    'bg-transparent hover:bg-grey-50': !disabled && !alt,
    'focus:bg-grey-50 focus:drop-shadow-focus-button': !disabled,
    'disabled:text-grey-200': disabled,
    //Tertiary - Blue
    'bg-grey-50': alt === 'blue',
    'border border-grey-500': icon,
    'hover:bg-grey-100': alt === 'blue' && !disabled
  });

  const buttonClass = classnames(
    `filter rounded-md font-inter transition-all outline-none ${
      className || ''
    }`,
    type === 'primary' ? primaryButtonClass : null,
    type === 'secondary' ? secondaryButtonClass : null,
    type === 'tertiary' ? tertiaryButtonClass : null,
    {
      'cursor-not-allowed': disabled,
      'flex flex-row items-center gap-x-2': type !== 'vertical',
      //Font Size
      'text-sm': (size === 'medium' || size === 'small') && alt !== 'vertical',
      'text-base': size === 'large' && alt !== 'vertical',
      'text-xs': alt === 'vertical',
      //Padding
      'p-2': size === 'small',
      'p-2.5': size === 'medium',
      'p-3': size === 'large'
    }
  );

  return (
    <button
      className={buttonClass}
      disabled={disabled}
      onClick={onClick}
      {...otherProps}>
      {children}
    </button>
  );
};

Button.propTypes = {
  /** Disabled */
  disabled: PropTypes.bool,
  /** Primary button: blue */
  primary: PropTypes.bool,
  /** Secondary button: white */
  secondary: PropTypes.bool,
  /** Secondary button: grey bg */
  tertiary: PropTypes.bool,
  /** Primary alt red */
  red: PropTypes.bool,
  //** Primary alt dark */
  dark: PropTypes.bool,
  //** Secondary alt vertical */
  vertical: PropTypes.bool,
  //** Tertiary alt blue */
  blue: PropTypes.bool,
  //** Button size small */
  small: PropTypes.bool,
  //** Button size large */
  large: PropTypes.bool,
  //React node children
  children: PropTypes.node,
  //onClick callback method when a click interaction occurs
  onClick: PropTypes.func,
  //If this is only an icon or not.
  icon: PropTypes.bool,
  // Tailwind Classes
  className: PropTypes.string
};

export default Button;
