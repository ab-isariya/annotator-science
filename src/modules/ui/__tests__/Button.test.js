import Button from '../Button';
import renderer from 'react-test-renderer';

import {ReactComponent as Plus} from '@svgs/PlusSign.svg';

describe('<Button/>', () => {
  it('Renders without crashing', () => {
    const comp = renderer.create(<Button></Button>).toJSON();

    expect(comp).toMatchSnapshot();
  });

  it("Renders it's children", () => {
    const comp = renderer.create(<Button>Hello</Button>).toJSON();

    expect(comp).toMatchSnapshot();
    expect(comp.children).not.toEqual(null);
  });

  it('Renders the primary styles', () => {
    const comp = renderer.create(<Button>Button</Button>).toJSON();
    const comp_withIcon = renderer
      .create(
        <Button>
          <Plus /> Button
        </Button>
      )
      .toJSON();
    const comp_iconOnly = renderer
      .create(
        <Button>
          <Plus />
        </Button>
      )
      .toJSON();
    const comp_disabled = renderer
      .create(
        <Button disabled>
          <Plus />
        </Button>
      )
      .toJSON();
    const comp_small = renderer
      .create(
        <Button small>
          <Plus />
        </Button>
      )
      .toJSON();
    const comp_large = renderer
      .create(
        <Button large>
          <Plus />
        </Button>
      )
      .toJSON();

    expect(comp).toMatchSnapshot();
    expect(comp_withIcon).toMatchSnapshot();
    expect(comp_iconOnly).toMatchSnapshot();
    expect(comp_disabled).toMatchSnapshot();
    expect(comp_small).toMatchSnapshot();
    expect(comp_large).toMatchSnapshot();
  });

  it('Renders the primary red styles', () => {
    const comp = renderer.create(<Button red>Button</Button>).toJSON();
    const comp_withIcon = renderer
      .create(
        <Button red>
          <Plus /> Button
        </Button>
      )
      .toJSON();
    const comp_iconOnly = renderer
      .create(
        <Button red>
          <Plus />
        </Button>
      )
      .toJSON();
    const comp_disabled = renderer
      .create(
        <Button red disabled>
          <Plus />
        </Button>
      )
      .toJSON();
    const comp_small = renderer
      .create(
        <Button red small>
          <Plus />
        </Button>
      )
      .toJSON();
    const comp_large = renderer
      .create(
        <Button red large>
          <Plus />
        </Button>
      )
      .toJSON();

    expect(comp).toMatchSnapshot();
    expect(comp_withIcon).toMatchSnapshot();
    expect(comp_iconOnly).toMatchSnapshot();
    expect(comp_disabled).toMatchSnapshot();
    expect(comp_small).toMatchSnapshot();
    expect(comp_large).toMatchSnapshot();
  });

  it('Renders the primary dark styles', () => {
    const comp = renderer.create(<Button dark>Button</Button>).toJSON();
    const comp_withIcon = renderer
      .create(
        <Button dark>
          <Plus /> Button
        </Button>
      )
      .toJSON();
    const comp_iconOnly = renderer
      .create(
        <Button dark>
          <Plus />
        </Button>
      )
      .toJSON();
    const comp_disabled = renderer
      .create(
        <Button dark disabled>
          <Plus />
        </Button>
      )
      .toJSON();
    const comp_small = renderer
      .create(
        <Button dark small>
          <Plus />
        </Button>
      )
      .toJSON();
    const comp_large = renderer
      .create(
        <Button dark large>
          <Plus />
        </Button>
      )
      .toJSON();

    expect(comp).toMatchSnapshot();
    expect(comp_withIcon).toMatchSnapshot();
    expect(comp_iconOnly).toMatchSnapshot();
    expect(comp_disabled).toMatchSnapshot();
    expect(comp_small).toMatchSnapshot();
    expect(comp_large).toMatchSnapshot();
  });

  it('Renders the secondary styles', () => {
    const comp = renderer.create(<Button secondary>Button</Button>).toJSON();
    const comp_withIcon = renderer
      .create(
        <Button secondary>
          <Plus /> Button
        </Button>
      )
      .toJSON();
    const comp_iconOnly = renderer
      .create(
        <Button secondary>
          <Plus />
        </Button>
      )
      .toJSON();
    const comp_disabled = renderer
      .create(
        <Button secondary disabled>
          <Plus />
        </Button>
      )
      .toJSON();
    const comp_small = renderer
      .create(
        <Button secondary small>
          <Plus />
        </Button>
      )
      .toJSON();
    const comp_large = renderer
      .create(
        <Button secondary large>
          <Plus />
        </Button>
      )
      .toJSON();

    expect(comp).toMatchSnapshot();
    expect(comp_withIcon).toMatchSnapshot();
    expect(comp_iconOnly).toMatchSnapshot();
    expect(comp_disabled).toMatchSnapshot();
    expect(comp_small).toMatchSnapshot();
    expect(comp_large).toMatchSnapshot();
  });

  it('Renders the secondary vertical styles', () => {
    const comp = renderer
      .create(
        <Button secondary vertical>
          Button
        </Button>
      )
      .toJSON();
    const comp_withIcon = renderer
      .create(
        <Button secondary vertical>
          <Plus /> Button
        </Button>
      )
      .toJSON();
    const comp_iconOnly = renderer
      .create(
        <Button secondary vertical>
          <Plus />
        </Button>
      )
      .toJSON();
    const comp_disabled = renderer
      .create(
        <Button secondary vertical disabled>
          <Plus />
        </Button>
      )
      .toJSON();
    const comp_small = renderer
      .create(
        <Button secondary vertical small>
          <Plus />
        </Button>
      )
      .toJSON();
    const comp_large = renderer
      .create(
        <Button secondary vertical large>
          <Plus />
        </Button>
      )
      .toJSON();

    expect(comp).toMatchSnapshot();
    expect(comp_withIcon).toMatchSnapshot();
    expect(comp_iconOnly).toMatchSnapshot();
    expect(comp_disabled).toMatchSnapshot();
    expect(comp_small).toMatchSnapshot();
    expect(comp_large).toMatchSnapshot();
  });

  it('Renders the tertiary styles', () => {
    const comp = renderer.create(<Button tertiary>Button</Button>).toJSON();
    const comp_withIcon = renderer
      .create(
        <Button tertiary>
          <Plus /> Button
        </Button>
      )
      .toJSON();
    const comp_iconOnly = renderer
      .create(
        <Button tertiary>
          <Plus />
        </Button>
      )
      .toJSON();
    const comp_disabled = renderer
      .create(
        <Button tertiary disabled>
          <Plus />
        </Button>
      )
      .toJSON();
    const comp_small = renderer
      .create(
        <Button tertiary small>
          <Plus />
        </Button>
      )
      .toJSON();
    const comp_large = renderer
      .create(
        <Button tertiary large>
          <Plus />
        </Button>
      )
      .toJSON();

    expect(comp).toMatchSnapshot();
    expect(comp_withIcon).toMatchSnapshot();
    expect(comp_iconOnly).toMatchSnapshot();
    expect(comp_disabled).toMatchSnapshot();
    expect(comp_small).toMatchSnapshot();
    expect(comp_large).toMatchSnapshot();
  });

  it('Renders the tertiary blue styles', () => {
    const comp = renderer
      .create(
        <Button tertiary blue>
          Button
        </Button>
      )
      .toJSON();
    const comp_withIcon = renderer
      .create(
        <Button tertiary blue>
          <Plus /> Button
        </Button>
      )
      .toJSON();
    const comp_iconOnly = renderer
      .create(
        <Button tertiary blue>
          <Plus />
        </Button>
      )
      .toJSON();
    const comp_disabled = renderer
      .create(
        <Button tertiary blue disabled>
          <Plus />
        </Button>
      )
      .toJSON();
    const comp_small = renderer
      .create(
        <Button tertiary blue small>
          <Plus />
        </Button>
      )
      .toJSON();
    const comp_large = renderer
      .create(
        <Button tertiary blue large>
          <Plus />
        </Button>
      )
      .toJSON();

    expect(comp).toMatchSnapshot();
    expect(comp_withIcon).toMatchSnapshot();
    expect(comp_iconOnly).toMatchSnapshot();
    expect(comp_disabled).toMatchSnapshot();
    expect(comp_small).toMatchSnapshot();
    expect(comp_large).toMatchSnapshot();
  });

  it('Renders with a custom classname', () => {
    const comp = renderer
      .create(<Button className="testClass"></Button>)
      .toJSON();

    expect(comp).toMatchSnapshot();
    expect(comp.props.className).toMatch(/testClass/i);
  });
});
