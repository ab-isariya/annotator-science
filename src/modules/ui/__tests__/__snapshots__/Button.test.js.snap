// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button/> Renders it's children 1`] = `
<button
  className="filter rounded-md font-inter transition-all outline-none  bg-blue-500 text-white hover:bg-blue-600 focus:bg-blue-600 focus:drop-shadow-focus-button flex flex-row items-center gap-x-2 text-sm p-2.5"
>
  Hello
</button>
`;

exports[`<Button/> Renders the primary dark styles 1`] = `
<button
  className="filter rounded-md font-inter transition-all outline-none  text-white bg-grey-600 hover:bg-grey-500 focus:bg-grey-500 focus:drop-shadow-focus-dark flex flex-row items-center gap-x-2 text-sm p-2.5"
>
  Button
</button>
`;

exports[`<Button/> Renders the primary dark styles 2`] = `
<button
  className="filter rounded-md font-inter transition-all outline-none  text-white bg-grey-600 hover:bg-grey-500 focus:bg-grey-500 focus:drop-shadow-focus-dark flex flex-row items-center gap-x-2 text-sm p-2.5"
>
  <svg>
    PlusSign.svg
  </svg>
   Button
</button>
`;

exports[`<Button/> Renders the primary dark styles 3`] = `
<button
  className="filter rounded-md font-inter transition-all outline-none  text-white bg-grey-600 hover:bg-grey-500 focus:bg-grey-500 focus:drop-shadow-focus-dark flex flex-row items-center gap-x-2 text-sm p-2.5"
>
  <svg>
    PlusSign.svg
  </svg>
</button>
`;

exports[`<Button/> Renders the primary dark styles 4`] = `
<button
  className="filter rounded-md font-inter transition-all outline-none  bg-grey-600 disabled:bg-grey-700 disabled:text-grey-400 cursor-not-allowed flex flex-row items-center gap-x-2 text-sm p-2.5"
  disabled={true}
>
  <svg>
    PlusSign.svg
  </svg>
</button>
`;

exports[`<Button/> Renders the primary dark styles 5`] = `
<button
  className="filter rounded-md font-inter transition-all outline-none  text-white bg-grey-600 hover:bg-grey-500 focus:bg-grey-500 focus:drop-shadow-focus-dark flex flex-row items-center gap-x-2 text-sm p-2"
>
  <svg>
    PlusSign.svg
  </svg>
</button>
`;

exports[`<Button/> Renders the primary dark styles 6`] = `
<button
  className="filter rounded-md font-inter transition-all outline-none  text-white bg-grey-600 hover:bg-grey-500 focus:bg-grey-500 focus:drop-shadow-focus-dark flex flex-row items-center gap-x-2 text-base p-3"
>
  <svg>
    PlusSign.svg
  </svg>
</button>
`;

exports[`<Button/> Renders the primary red styles 1`] = `
<button
  className="filter rounded-md font-inter transition-all outline-none  text-white bg-red-600 hover:bg-red-700 focus:bg-red-700 focus:drop-shadow-focus-red flex flex-row items-center gap-x-2 text-sm p-2.5"
>
  Button
</button>
`;

exports[`<Button/> Renders the primary red styles 2`] = `
<button
  className="filter rounded-md font-inter transition-all outline-none  text-white bg-red-600 hover:bg-red-700 focus:bg-red-700 focus:drop-shadow-focus-red flex flex-row items-center gap-x-2 text-sm p-2.5"
>
  <svg>
    PlusSign.svg
  </svg>
   Button
</button>
`;

exports[`<Button/> Renders the primary red styles 3`] = `
<button
  className="filter rounded-md font-inter transition-all outline-none  text-white bg-red-600 hover:bg-red-700 focus:bg-red-700 focus:drop-shadow-focus-red flex flex-row items-center gap-x-2 text-sm p-2.5"
>
  <svg>
    PlusSign.svg
  </svg>
</button>
`;

exports[`<Button/> Renders the primary red styles 4`] = `
<button
  className="filter rounded-md font-inter transition-all outline-none  disabled:bg-grey-100 disabled:text-grey-300 bg-red-600 cursor-not-allowed flex flex-row items-center gap-x-2 text-sm p-2.5"
  disabled={true}
>
  <svg>
    PlusSign.svg
  </svg>
</button>
`;

exports[`<Button/> Renders the primary red styles 5`] = `
<button
  className="filter rounded-md font-inter transition-all outline-none  text-white bg-red-600 hover:bg-red-700 focus:bg-red-700 focus:drop-shadow-focus-red flex flex-row items-center gap-x-2 text-sm p-2"
>
  <svg>
    PlusSign.svg
  </svg>
</button>
`;

exports[`<Button/> Renders the primary red styles 6`] = `
<button
  className="filter rounded-md font-inter transition-all outline-none  text-white bg-red-600 hover:bg-red-700 focus:bg-red-700 focus:drop-shadow-focus-red flex flex-row items-center gap-x-2 text-base p-3"
>
  <svg>
    PlusSign.svg
  </svg>
</button>
`;

exports[`<Button/> Renders the primary styles 1`] = `
<button
  className="filter rounded-md font-inter transition-all outline-none  bg-blue-500 text-white hover:bg-blue-600 focus:bg-blue-600 focus:drop-shadow-focus-button flex flex-row items-center gap-x-2 text-sm p-2.5"
>
  Button
</button>
`;

exports[`<Button/> Renders the primary styles 2`] = `
<button
  className="filter rounded-md font-inter transition-all outline-none  bg-blue-500 text-white hover:bg-blue-600 focus:bg-blue-600 focus:drop-shadow-focus-button flex flex-row items-center gap-x-2 text-sm p-2.5"
>
  <svg>
    PlusSign.svg
  </svg>
   Button
</button>
`;

exports[`<Button/> Renders the primary styles 3`] = `
<button
  className="filter rounded-md font-inter transition-all outline-none  bg-blue-500 text-white hover:bg-blue-600 focus:bg-blue-600 focus:drop-shadow-focus-button flex flex-row items-center gap-x-2 text-sm p-2.5"
>
  <svg>
    PlusSign.svg
  </svg>
</button>
`;

exports[`<Button/> Renders the primary styles 4`] = `
<button
  className="filter rounded-md font-inter transition-all outline-none  bg-blue-500 disabled:bg-grey-100 disabled:text-grey-300 cursor-not-allowed flex flex-row items-center gap-x-2 text-sm p-2.5"
  disabled={true}
>
  <svg>
    PlusSign.svg
  </svg>
</button>
`;

exports[`<Button/> Renders the primary styles 5`] = `
<button
  className="filter rounded-md font-inter transition-all outline-none  bg-blue-500 text-white hover:bg-blue-600 focus:bg-blue-600 focus:drop-shadow-focus-button flex flex-row items-center gap-x-2 text-sm p-2"
>
  <svg>
    PlusSign.svg
  </svg>
</button>
`;

exports[`<Button/> Renders the primary styles 6`] = `
<button
  className="filter rounded-md font-inter transition-all outline-none  bg-blue-500 text-white hover:bg-blue-600 focus:bg-blue-600 focus:drop-shadow-focus-button flex flex-row items-center gap-x-2 text-base p-3"
>
  <svg>
    PlusSign.svg
  </svg>
</button>
`;

exports[`<Button/> Renders the secondary styles 1`] = `
<button
  className="filter rounded-md font-inter transition-all outline-none  bg-white border border-grey-500 hover:bg-grey-50 focus:bg-grey-50 focus:drop-shadow-focus-button flex flex-row items-center gap-x-2 text-sm p-2.5"
>
  Button
</button>
`;

exports[`<Button/> Renders the secondary styles 2`] = `
<button
  className="filter rounded-md font-inter transition-all outline-none  bg-white border border-grey-500 hover:bg-grey-50 focus:bg-grey-50 focus:drop-shadow-focus-button flex flex-row items-center gap-x-2 text-sm p-2.5"
>
  <svg>
    PlusSign.svg
  </svg>
   Button
</button>
`;

exports[`<Button/> Renders the secondary styles 3`] = `
<button
  className="filter rounded-md font-inter transition-all outline-none  bg-white border border-grey-500 hover:bg-grey-50 focus:bg-grey-50 focus:drop-shadow-focus-button flex flex-row items-center gap-x-2 text-sm p-2.5"
>
  <svg>
    PlusSign.svg
  </svg>
</button>
`;

exports[`<Button/> Renders the secondary styles 4`] = `
<button
  className="filter rounded-md font-inter transition-all outline-none  bg-white border disabled:text-grey-200 disabled:border-grey-200 cursor-not-allowed flex flex-row items-center gap-x-2 text-sm p-2.5"
  disabled={true}
>
  <svg>
    PlusSign.svg
  </svg>
</button>
`;

exports[`<Button/> Renders the secondary styles 5`] = `
<button
  className="filter rounded-md font-inter transition-all outline-none  bg-white border border-grey-500 hover:bg-grey-50 focus:bg-grey-50 focus:drop-shadow-focus-button flex flex-row items-center gap-x-2 text-sm p-2"
>
  <svg>
    PlusSign.svg
  </svg>
</button>
`;

exports[`<Button/> Renders the secondary styles 6`] = `
<button
  className="filter rounded-md font-inter transition-all outline-none  bg-white border border-grey-500 hover:bg-grey-50 focus:bg-grey-50 focus:drop-shadow-focus-button flex flex-row items-center gap-x-2 text-base p-3"
>
  <svg>
    PlusSign.svg
  </svg>
</button>
`;

exports[`<Button/> Renders the secondary vertical styles 1`] = `
<button
  className="filter rounded-md font-inter transition-all outline-none  bg-white border border-grey-500 hover:bg-grey-50 focus:bg-grey-50 focus:drop-shadow-focus-button rounded-md px-1.5 py-1.5 flex flex-col items-center justify-center text-xs gap-y-1 flex flex-row items-center gap-x-2 text-xs p-2.5"
>
  Button
</button>
`;

exports[`<Button/> Renders the secondary vertical styles 2`] = `
<button
  className="filter rounded-md font-inter transition-all outline-none  bg-white border border-grey-500 hover:bg-grey-50 focus:bg-grey-50 focus:drop-shadow-focus-button rounded-md px-1.5 py-1.5 flex flex-col items-center justify-center text-xs gap-y-1 flex flex-row items-center gap-x-2 text-xs p-2.5"
>
  <svg>
    PlusSign.svg
  </svg>
   Button
</button>
`;

exports[`<Button/> Renders the secondary vertical styles 3`] = `
<button
  className="filter rounded-md font-inter transition-all outline-none  bg-white border border-grey-500 hover:bg-grey-50 focus:bg-grey-50 focus:drop-shadow-focus-button rounded-md px-1.5 py-1.5 flex flex-col items-center justify-center text-xs gap-y-1 flex flex-row items-center gap-x-2 text-xs p-2.5"
>
  <svg>
    PlusSign.svg
  </svg>
</button>
`;

exports[`<Button/> Renders the secondary vertical styles 4`] = `
<button
  className="filter rounded-md font-inter transition-all outline-none  bg-white border disabled:text-grey-200 disabled:border-grey-200 rounded-md px-1.5 py-1.5 flex flex-col items-center justify-center text-xs gap-y-1 cursor-not-allowed flex flex-row items-center gap-x-2 text-xs p-2.5"
  disabled={true}
>
  <svg>
    PlusSign.svg
  </svg>
</button>
`;

exports[`<Button/> Renders the secondary vertical styles 5`] = `
<button
  className="filter rounded-md font-inter transition-all outline-none  bg-white border border-grey-500 hover:bg-grey-50 focus:bg-grey-50 focus:drop-shadow-focus-button rounded-md px-1.5 py-1.5 flex flex-col items-center justify-center text-xs gap-y-1 flex flex-row items-center gap-x-2 text-xs p-2"
>
  <svg>
    PlusSign.svg
  </svg>
</button>
`;

exports[`<Button/> Renders the secondary vertical styles 6`] = `
<button
  className="filter rounded-md font-inter transition-all outline-none  bg-white border border-grey-500 hover:bg-grey-50 focus:bg-grey-50 focus:drop-shadow-focus-button rounded-md px-1.5 py-1.5 flex flex-col items-center justify-center text-xs gap-y-1 flex flex-row items-center gap-x-2 text-xs p-3"
>
  <svg>
    PlusSign.svg
  </svg>
</button>
`;

exports[`<Button/> Renders the tertiary blue styles 1`] = `
<button
  className="filter rounded-md font-inter transition-all outline-none  focus:bg-grey-50 focus:drop-shadow-focus-button bg-grey-50 hover:bg-grey-100 flex flex-row items-center gap-x-2 text-sm p-2.5"
>
  Button
</button>
`;

exports[`<Button/> Renders the tertiary blue styles 2`] = `
<button
  className="filter rounded-md font-inter transition-all outline-none  focus:bg-grey-50 focus:drop-shadow-focus-button bg-grey-50 hover:bg-grey-100 flex flex-row items-center gap-x-2 text-sm p-2.5"
>
  <svg>
    PlusSign.svg
  </svg>
   Button
</button>
`;

exports[`<Button/> Renders the tertiary blue styles 3`] = `
<button
  className="filter rounded-md font-inter transition-all outline-none  focus:bg-grey-50 focus:drop-shadow-focus-button bg-grey-50 hover:bg-grey-100 flex flex-row items-center gap-x-2 text-sm p-2.5"
>
  <svg>
    PlusSign.svg
  </svg>
</button>
`;

exports[`<Button/> Renders the tertiary blue styles 4`] = `
<button
  className="filter rounded-md font-inter transition-all outline-none  disabled:text-grey-200 bg-grey-50 cursor-not-allowed flex flex-row items-center gap-x-2 text-sm p-2.5"
  disabled={true}
>
  <svg>
    PlusSign.svg
  </svg>
</button>
`;

exports[`<Button/> Renders the tertiary blue styles 5`] = `
<button
  className="filter rounded-md font-inter transition-all outline-none  focus:bg-grey-50 focus:drop-shadow-focus-button bg-grey-50 hover:bg-grey-100 flex flex-row items-center gap-x-2 text-sm p-2"
>
  <svg>
    PlusSign.svg
  </svg>
</button>
`;

exports[`<Button/> Renders the tertiary blue styles 6`] = `
<button
  className="filter rounded-md font-inter transition-all outline-none  focus:bg-grey-50 focus:drop-shadow-focus-button bg-grey-50 hover:bg-grey-100 flex flex-row items-center gap-x-2 text-base p-3"
>
  <svg>
    PlusSign.svg
  </svg>
</button>
`;

exports[`<Button/> Renders the tertiary styles 1`] = `
<button
  className="filter rounded-md font-inter transition-all outline-none  bg-transparent hover:bg-grey-50 focus:bg-grey-50 focus:drop-shadow-focus-button flex flex-row items-center gap-x-2 text-sm p-2.5"
>
  Button
</button>
`;

exports[`<Button/> Renders the tertiary styles 2`] = `
<button
  className="filter rounded-md font-inter transition-all outline-none  bg-transparent hover:bg-grey-50 focus:bg-grey-50 focus:drop-shadow-focus-button flex flex-row items-center gap-x-2 text-sm p-2.5"
>
  <svg>
    PlusSign.svg
  </svg>
   Button
</button>
`;

exports[`<Button/> Renders the tertiary styles 3`] = `
<button
  className="filter rounded-md font-inter transition-all outline-none  bg-transparent hover:bg-grey-50 focus:bg-grey-50 focus:drop-shadow-focus-button flex flex-row items-center gap-x-2 text-sm p-2.5"
>
  <svg>
    PlusSign.svg
  </svg>
</button>
`;

exports[`<Button/> Renders the tertiary styles 4`] = `
<button
  className="filter rounded-md font-inter transition-all outline-none  disabled:text-grey-200 cursor-not-allowed flex flex-row items-center gap-x-2 text-sm p-2.5"
  disabled={true}
>
  <svg>
    PlusSign.svg
  </svg>
</button>
`;

exports[`<Button/> Renders the tertiary styles 5`] = `
<button
  className="filter rounded-md font-inter transition-all outline-none  bg-transparent hover:bg-grey-50 focus:bg-grey-50 focus:drop-shadow-focus-button flex flex-row items-center gap-x-2 text-sm p-2"
>
  <svg>
    PlusSign.svg
  </svg>
</button>
`;

exports[`<Button/> Renders the tertiary styles 6`] = `
<button
  className="filter rounded-md font-inter transition-all outline-none  bg-transparent hover:bg-grey-50 focus:bg-grey-50 focus:drop-shadow-focus-button flex flex-row items-center gap-x-2 text-base p-3"
>
  <svg>
    PlusSign.svg
  </svg>
</button>
`;

exports[`<Button/> Renders with a custom classname 1`] = `
<button
  className="filter rounded-md font-inter transition-all outline-none testClass bg-blue-500 text-white hover:bg-blue-600 focus:bg-blue-600 focus:drop-shadow-focus-button flex flex-row items-center gap-x-2 text-sm p-2.5"
/>
`;

exports[`<Button/> Renders without crashing 1`] = `
<button
  className="filter rounded-md font-inter transition-all outline-none  bg-blue-500 text-white hover:bg-blue-600 focus:bg-blue-600 focus:drop-shadow-focus-button flex flex-row items-center gap-x-2 text-sm p-2.5"
/>
`;
